### DISCUSSION TITLE
Refining the block inserter

### DISCUSSION THREAD

**Discussion by jameskoster** [WordPress/gutenberg#35800] [Ideas]
Currently there are a few problems that harm the overall usability of the block inserter:

1. Install a few block plugins and it becomes quite painful to navigate around – there's lots of scrolling involved which harms what little hierarchy exists. 
2. Un-intuitive categories... we have “Theme” and “Design” which feel a little at-odds with one another. "Widgets" is a paradigm with a finite lifespan. Embeds is a large category that could potentially use some internal organisation.
3. All blocks are available, all the time. So I'm able to add something like a "Post Content" block to a post or page which obviously doesn't work. I'm also able to add the More block to a template, which also does not work. 

Here are some ideas to address these issues:

1. Re-organise the blocks in to categories that are more intuitive to browse.
2. Use the drilldown pattern from global styles to make the initial view more useful and less overwhelming.
3. Add a "Frequently used" section which is visible when the Inserter is opened.
4. Contextualise the Inserter so that template blocks aren't available in the post editor, and vice versa.

<img width="811" alt="Screenshot 2021-10-20 at 11 57 37" src="https://user-images.githubusercontent.com/846565/138080806-4e9b0197-bc92-4d79-8759-05ba10fa0136.png">

I made a prototype to get a feel for how these concepts combine. [Take a look](https://cdpn.io/jameskoster/debug/QWgomXv). Note that this is the Inserter you would see in the Post editor. In the template editor you would see a slightly different set of block categories, but I didn't make a prototype for that.

The category names are just a first take – we can certainly do better. The important thing in that regard is to concentrate on the block groupings themselves to determine whether or not they feel right.

---

I acknowledge there's a lot going on here. It may not all be good. Maybe none of it is :)

**Comment by annezazu** [comment]
Thanks for this exploration! I totally agree with your take on the current experience and its pain points. Some general feedback/ideas:
- Can't tell if I'd want a frequently used or favorites section :D 
- I want to see what this view would turn into upon searching (guessing it would return to the current experience?). 
- I worry if this looks too similar to global styles that folks might get confused by the different sidebars and think that they can customize the appearance of these blocks from the Inserter. 
- I don't know how much of an impact it'll make not being able to see all of the different block options in terms of what users think is possible with blocks. At the same time, I think the current experience is a bit too overwhelming regardless. 
- I worry about the additional clicks this will all require compared to the current browsing experience. 
Hope these quick thoughts are helpful!

**Comment by talldan** [comment]
> Can't tell if I'd want a frequently used or favorites section :D

Favorites or being able to 'pin' a block is a very requested feature! I think there are some issues out there with lots of comments asking for this.

Frequently used is already a feature, but it has to be enabled in the editor preferences in the 'blocks' section. A lot of folks don't seem to know this exists.
